browser/ui/webui/settings/brave_privacy_handler.cc

```diff
@@ -69,27 +78,58 @@ void BravePrivacyHandler::AddLoadTimeData(content::WebUIDataSource* data_source,
 #endif
 }
 
-#if BUILDFLAG(BRAVE_P3A_ENABLED)
-void BravePrivacyHandler::SetP3AEnabled(const base::ListValue* args) {
+void BravePrivacyHandler::SetLocalStateBooleanEnabled(
+    const std::string& path,
+    const base::ListValue* args) {
   CHECK_EQ(args->GetSize(), 1U);
 
   bool enabled;
   args->GetBoolean(0, &enabled);
 
   PrefService* local_state = g_browser_process->local_state();
-  local_state->SetBoolean(brave::kP3AEnabled, enabled);
+  local_state->SetBoolean(path, enabled);
 }
 
-void BravePrivacyHandler::GetP3AEnabled(const base::ListValue* args) {
+void BravePrivacyHandler::GetLocalStateBooleanEnabled(
+    const std::string& path,
+    const base::ListValue* args) {
   CHECK_EQ(args->GetSize(), 1U);
 
   PrefService* local_state = g_browser_process->local_state();
-  bool enabled = local_state->GetBoolean(brave::kP3AEnabled);
+  bool enabled = local_state->GetBoolean(path);
 
   AllowJavascript();
   ResolveJavascriptCallback(args->GetList()[0].Clone(), base::Value(enabled));
 }
 
+void BravePrivacyHandler::SetStatsUsagePingEnabled(
+    const base::ListValue* args) {
+  SetLocalStateBooleanEnabled(kStatsReportingEnabled, args);
+}
+
+void BravePrivacyHandler::GetStatsUsagePingEnabled(
+    const base::ListValue* args) {
+  GetLocalStateBooleanEnabled(kStatsReportingEnabled, args);
+}
+
+#if BUILDFLAG(BRAVE_P3A_ENABLED)
+void BravePrivacyHandler::SetP3AEnabled(const base::ListValue* args) {
+  SetLocalStateBooleanEnabled(brave::kP3AEnabled, args);
+}
+
+void BravePrivacyHandler::GetP3AEnabled(const base::ListValue* args) {
+  GetLocalStateBooleanEnabled(brave::kP3AEnabled, args);
+}
+
+void BravePrivacyHandler::OnStatsUsagePingEnabledChanged() {
+  if (IsJavascriptAllowed()) {
+    PrefService* local_state = g_browser_process->local_state();
+    bool enabled = local_state->GetBoolean(brave::kP3AEnabled);
```

Comment:
Pref name should be `kStatsReportingEnabled` instead of  `brave::kP3AEnabled`?
and is it correct that this method is inside `BRAVE_P3A_ENABLED` guard?

Reply:
fixed